pylearn2/datasets/dense_design_matrix.py

```diff
@@ -264,13 +264,14 @@ def _check_labels(self):
             assert self.y.ndim <= 2
             assert np.all(self.y < self.y_labels)
 
-    @functools.wraps(Dataset.iterator)
+    # Cannot wrap the iterator method cause it is abstract
+    # @functools.wraps(Dataset.iterator)
```

Comment:
Keeping the `functools.wraps` results in an error because it tries to call `Dataset.iterator`, which is an abstract method. 

You should either get rid of the abstract decorator, falling back in having a fake interface that is really just a class and as such can be instantiated, its method can be called, etc, .., or find a better way to inherit docstrings. 

Isn't there a way to specify it to the documentation generation tool? That would be the best option in my opinion. Another option is to add \__doc\__ += Dataset.iterator.\__doc\__ or use a more sophisticated way like writing your own decorator that inherits the docstring without actually calling the inherited method.